signature/sign_test.go

```diff
@@ -134,6 +139,126 @@ func TestSignVerify(t *testing.T) {
 	}
 }
 
+var _ crypto.Signer = MockCryptoSigner{}
+
+type MockCryptoSigner struct {
```

Comment:
My only quibble here (very minor) is I would call this a _fake_ instead of a _mock_, since it lacks the ability to check whether its methods were called (a good thing).